src/enumerable.cr

```diff
@@ -281,22 +281,22 @@ module Enumerable(T)
   #
   # By default, a new array is created and yielded for each consecutive slice of elements.
   # * If *reuse* is given, the array can be reused
-  # * If *reuse* is an `Array`, this array will be reused
-  # * If *reuse* is truthy, the method will create a new array and reuse it.
+  # * If *reuse* is `true`, the method will create a new array and reuse it.
+  # * If *reuse* is truthy, the passed `Array` or `Deque` will be used.
```

Comment:
An integer would also be truthy, but can't be passed.
I'd suggest something along:
```
If *reuse*  is an instance of `Array`, `Deque` or a similar collection type (implementing `#<<`, `#shift` and `#size`) it will be used.
```

We should probably have an abstract module for a generic list interface.

Reply:
Well, updated, and that list interface is way out of scope.